pkg/controlplane/apiserver/config.go

```diff
@@ -147,13 +147,15 @@ func BuildGenericConfig(
 		return
 	}
 
-	genericConfig.Authorization.Authorizer, genericConfig.RuleResolver, err = BuildAuthorizer(s, genericConfig.EgressSelector, versionedInformers)
-	if err != nil {
-		lastErr = fmt.Errorf("invalid authorization config: %v", err)
-		return
-	}
-	if !sets.NewString(s.Authorization.Modes...).Has(modes.ModeRBAC) {
-		genericConfig.DisabledPostStartHooks.Insert(rbacrest.PostStartHookName)
+	if s.Authorization != nil {
+		genericConfig.Authorization.Authorizer, genericConfig.RuleResolver, err = BuildAuthorizer(s, genericConfig.EgressSelector, versionedInformers)
```

Comment:
godoc update that it can be `nil`.

When looking at https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go#L53-L56 this indicates that authorization is considered optional so this should be fine.

Reply:
Went another route: have fixed all those option structs in controlplane/apiserver that panic'ed when nil. They are supposed to be optional. Have also documented that.